scripts/push-docker-images.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+# Push our docker images to a registry
+set -xeuo pipefail
+
+# Manually set the registry to a docker registry where you have push access to test this script.
+REGISTRY=${REGISTRY:-gomods/}
+
+# Try to use the travis variables before hitting git because travis does shallow clones for performance reasons.
+VERSION=${VERSION:-$TRAVIS_TAG}
+if [[ -z "$VERSION" ]]; then
+    echo "defaulting VERSION using git..."
+    VERSION=`git describe --tags --abbrev=7 --dirty`
+fi
+
+BRANCH=${BRANCH:-$TRAVIS_BRANCH}
+if [[-z "$BRANCH" ]]; then
+    echo "defaulting BRANCH using git..."
+    BRANCH=`git rev-parse --abbrev-ref HEAD`
+    if [[ "$BRANCH" == "HEAD" ]]; then
+        BRANCH=${VERSION} # Make our branch var work like travis_branch
+    fi
+fi
+
+# mutable tag is the docker image tag that we will reuse between pushes, it is not a stable tag like a SHA or version.
+MUTABLE_TAG=${MUTABLE_TAG:-} # defaulted below based on the branch
+if [[ -z "$MUTABLE_TAG" ]]; then
```

Comment:
I prefer your original implementation using case statement.  I thought it was simpler that way